account/templates/account/dashboard.html

```diff
@@ -37,28 +275,98 @@
     }));
 
 
-    console.log(data)
-    console.log('I am calling someone')
+    //jQuery('#callingmodal').modal('show');
+    $('#callingmodal').on('show.bs.modal', function (e) {
+        $('#callingdetails').text("Calling " + data.name)
+    })
+    
+    $('#callingmodal').modal('show')
+
+    
 
 }
 
 
 function processCall(userName) {
-    sendCall({name: userName,rtcMessage: "sessionDescription"})
-   
+    peerConnection.createOffer((sessionDescription) => {
+        peerConnection.setLocalDescription(sessionDescription);
+        sendCall({
+            name: userName,
+            rtcMessage: sessionDescription
+        })
+    }, (error) => {
+        console.log("Error");
+    });
 }
 
 
+
 const onNewCall = (data) =>{
         //when other called you
         //show answer button
 
         otherUser = data.caller;
         remoteRTCMessage = data.rtcMessage
 
-        console.log('I am receiving call')
+        console.log('I am receiving call from %s', data.caller)
+
+        $('#receivingmodal').on('show.bs.modal', function (e) {
+            $('#receivingdetails').text(data.caller + " is Calling You")
+        })
+    
+        $('#receivingmodal').modal('show')
+        
     
+}
+
+
+function callProgress() {
+    callInProgress = true;
+}
+
+
+const onCallAnswered = (data) =>{
+        //when other accept our call
+        remoteRTCMessage = data.rtcMessage
+        peerConnection.setRemoteDescription(new RTCSessionDescription(remoteRTCMessage));
+
+        
+
+        console.log("Call Started. They Answered");
+        // console.log(pc);
+
+        callProgress()
+}
+
+
+const onICECandidate = (data) =>{
+        // console.log(data);
+        console.log("GOT ICE candidate");
+
+        let message = data.rtcMessage
+
+        let candidate = new RTCIceCandidate({
+            sdpMLineIndex: message.label,
+            candidate: message.candidate
+        });
+
+        if (peerConnection) {
+            console.log("ICE candidate Added");
+            peerConnection.addIceCandidate(candidate);
+        } else {
+            console.log("ICE candidate Pushed");
+            iceCandidatesFromCaller.push(candidate);
+        }
+
+}
+
+
+window.onbeforeunload = function () {
+    if (callInProgress) {
+        stop();
     }
+};
+
 
 
 $(document).ready(function(){
```

Comment:
line 373 - 377 is duplicate to line 98 - 102. Line 373 - 377 can be removed.

Reply:
I removed this piece of code and repushed.